wire-ios-data-model/MessagingService.swift

```diff
@@ -0,0 +1,120 @@
+////
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class MessagingService {
+
+    let proteusProvider: ProteusProvider
+    let mlsService: MLSServiceInterface?
+    let managedObjectContext: NSManagedObjectContext
+
+    init(proteusProvider: ProteusProvider, mlsService: MLSServiceInterface?, managedObjectContext: NSManagedObjectContext) {
+        self.proteusProvider = proteusProvider
+        self.mlsService = mlsService
+        self.managedObjectContext = managedObjectContext
+    }
+}
+
+public class FingerprintUseCase {
+
+    let messagingService: MessagingService
```

Comment:
`Suggestion:` Use interface instead of concrete classes for dependency which would help us writing better unit tests.